core/src/main/java/org/mule/transformer/types/DataTypeFactory.java

```diff
@@ -41,11 +41,11 @@
     public static final DataType<String> RSS_STRING = new SimpleDataType<>(String.class, MimeTypes.RSS);
 
     //Common Java types
-    public static final DataType<String> STRING = new SimpleDataType<>(String.class);
-    public static final DataType<String> OBJECT = new SimpleDataType<>(Object.class);
-    public static final DataType<String> BYTE_ARRAY = new SimpleDataType<>(byte[].class);
-    public static final DataType<String> INPUT_STREAM = new SimpleDataType<>(InputStream.class);
-    public static final DataType<String> MULE_MESSAGE = new SimpleDataType<>(MuleMessage.class);
+    public static final DataType<String> STRING = new SimpleDataType<String>(String.class);
```

Comment:
Can't remember if this was strictly needed, but DataType if created with generics should use correct type and not Object right?